### PR TITLE
extend node configuration

### PR DESCRIPTION
**Pull request by NazariiDenha**
Extend node configuration with: L1Endpoint, L1Confirmations, L1DeploymentBlock

https://www.notion.so/scrollzkp/SPLIT-Use-Sequencer-to-include-L1-messages-243d772e13bd40439fd17c9ed47d5781?pvs=4#c4e5acedb65d4968bc4e7a8d01526117

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -790,6 +790,20 @@ var (
...
+		Name:  "l1.addr",
+		Usage: "Endpoint of L1 HTTP-RPC server",
+	}
+	L1ConfirmationsFlag = cli.IntFlag{
+		Name:  "l1.confirmations",
+		Usage: "Number of confirmations on L1 needed for finalization",
+	}
+	L1DeploymentBlockFlag = cli.StringFlag{
+		Name:  "l1.deployment.block",
+		Usage: "Hash od block where bridge contract is deployed on L1",
```
Comment (Thegaram) [inline cmd/utils/flags.go]:
```suggestion
		Usage: "Hash of block where bridge contract is deployed on L1",
```

Comment (Thegaram) [reply cmd/utils/flags.go]: Why use hash? I think it's fine to use the block number

File: node/config.go [hunk 2]
```diff
@@ -190,6 +190,13 @@ type Config struct {
 
 	// AllowUnprotectedTxs allows non EIP-155 protected transactions to be send over RPC.
 	AllowUnprotectedTxs bool `toml:",omitempty"`
+
+	// Endpoint of L1 HTTP-RPC server
+	L1Endpoint string `toml:",omitempty"`
+	// Number of confirmations on L1 needed for finalization
+	L1Confirmations int `toml:",omitempty"`
```
Comment (Thegaram) [inline node/config.go]: Can we also support `safe` and `finalized` as we do in scroll? There we use `rpc.BlockNumber` for this (but I think that only accepts hex for numbers, might be inconvenient to use)

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -1242,6 +1259,38 @@ func SetNodeConfig(ctx *cli.Context, cfg *node.Config) {
 	}
 }
 
+func unmarshalBlockNumber(input string) rpc.BlockNumber {
```
Comment (Thegaram) [inline cmd/utils/flags.go:1262]: Can we just use [`BlockNumber.UnmarshalJSON`](https://github.com/scroll-tech/go-ethereum/blob/develop/rpc/types.go#L74)? And using a default is dangerous, let's just return the parse error to the user.

Comment (NazariiDenha) [reply cmd/utils/flags.go:1262]: `umarshaljson` doesn't support usual numbers, only hex

Comment (Thegaram) [reply cmd/utils/flags.go:1262]: TODO:
- Raise error on invalid hex/number passed
- Figure out what the default value is if user does not specify

**Pull request merged**